web/src/test/java/com/epam/rft/atsy/web/controllers/rest/CandidateApplicationControllerTest.java

```diff
@@ -0,0 +1,180 @@
+
+  @Mock
+  private StatesHistoryService statesHistoryService;
+
+  @Mock
+  private MessageSource messageSource;
+
+  @InjectMocks
+  private CandidateApplicationController candidateApplicationController;
+
+  private CandidateApplicationDTO developerApplicationDto;
+
+  private CandidateApplicationDTO sysadminApplicationDto;
+
+  private Collection<CandidateApplicationDTO> candidateApplicationCollection;
+
+  @Override
+  protected Object[] controllersUnderTest() {
+    return new Object[]{candidateApplicationController};
+  }
+
+  @Before
+  public void setUpTestData() {
+    developerApplicationDto =
+        CandidateApplicationDTO.builder().lastStateId(LAST_STATE_ID)
+            .applicationId(APPLICATION_ID)
+            .positionName(DEVELOPER_POSITION).creationDate(CREATION_DATE)
+            .modificationDate(MODIFICATION_DATE)
+            .stateType(RAW_STATE_TYPE)
+            .build();
+
+    sysadminApplicationDto =
+        CandidateApplicationDTO.builder().lastStateId(LAST_STATE_ID)
+            .applicationId(APPLICATION_ID)
+            .positionName(SYSADMIN_POSITION).creationDate(CREATION_DATE)
+            .modificationDate(MODIFICATION_DATE)
+            .stateType(RAW_STATE_TYPE)
+            .build();
+
+    candidateApplicationCollection = Arrays.asList(developerApplicationDto, sysadminApplicationDto);
+
+    given(messageSource.getMessage(Matchers.anyString(), Matchers.any(Object[].class),
+        Matchers.any(Locale.class))).willReturn(LOCALIZED_STATE_TYPE);
+  }
+
+  @Test
+  public void loadApplicationsShouldRespondWithEmptyCollectionWhenThereAreNoApplications()
+      throws Exception {
+    given(statesHistoryService.getCandidateApplicationsByCandidateId(CANDIDATE_ID)).willReturn(
+        Collections.emptyList());
+
+    mockMvc.perform(
+        get(REQUEST_URL + CANDIDATE_ID.toString()).accept(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$").isArray())
+        .andExpect(jsonPath("$").isEmpty());
+
+    then(statesHistoryService).should().getCandidateApplicationsByCandidateId(CANDIDATE_ID);
+
+    verifyZeroInteractions(messageSource);
+  }
+
+  @Test
+  public void loadApplicationsShouldRespondWithSingleElementArrayWhenThereIsOnlyOneApplication()
+      throws Exception {
+    given(statesHistoryService.getCandidateApplicationsByCandidateId(CANDIDATE_ID))
+        .willReturn(Collections.singletonList(developerApplicationDto));
+
+    mockMvc.perform(
+        get(REQUEST_URL + CANDIDATE_ID.toString()).accept(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$").isArray())
+        .andExpect(jsonPath("$[0]").exists())
+        .andExpect(jsonPath("$[0].lastStateId", equalTo(LAST_STATE_ID.intValue())))
```

Comment:
$[0], lastStateId, positionName, etc. are used quite a lot of times. These could be extracted and concatenated at the places of the usage.